tools/update_unicode_iso8859_tables.dart

```diff
@@ -0,0 +1,300 @@
+    // If no change, no write.
+    stderr.writeln('No changes');
+    return;
+  }
+  var replacement = replacementString(declarations);
+  // Guess file's newline, change to '\r\n' if necessary.
+  if (Platform.isWindows && tableFileContent.contains('\r\n')) {
+    replacement = replacement.replaceAll('\n', '\r\n');
+  }
+  tableFileContent = tableFileContent.replaceRange(
+      generatedContentMatch.start, generatedContentMatch.end, replacement);
+  tableFile.writeAsStringSync(tableFileContent);
+  print('Written: $tableFile');
+}
+
+/// Shared HTTP client used by [fetchTable], initialized if needed.
+///
+/// Closed by [generateTables] when done fetching.
+HttpClient? httpClient;
+
+/// Fetches table file content, parses and builds constant string declarations.
+Future<String> generateTables() async {
+  // Used for caching fetched files.
+  // Useful if running the script multiple times.
+  var cacheDir = Directory(cachePath);
+  if (!cacheDir.existsSync()) cacheDir.createSync(recursive: true);
+
+  var buffer = StringBuffer();
+  // For each ISO-8859-n, n > 1, fetch the table,
+  // and generate a constant string corresponding to the mapping
+  // into `buffer`.
+  // There is no ISO-8859-12. Don't include ISO-8859-1,
+  // since it's already included in `dart:convert`.
+  for (var i = 2; i <= 16; i++) {
+    if (i == 12) continue; // No ISO-8859-12.
+    await generateTableFor(buffer, i);
+  }
+
+  httpClient?.close();
+  return buffer.toString();
+}
+
+/// Generates the constant string for a single ISO-8859 code page.
+Future<void> generateTableFor(StringSink buffer, int isoNumber) async {
+  var (uri, content) = await fetchTable(isoNumber);
+  var text = parseTable(isoNumber, content);
+  buffer
+    ..writeln()
+    ..writeln('// Characters of ISO-8859-$isoNumber.')
+    ..write('const iso8859_$isoNumber = // From $uri');
+  // Write 8 chars per line, as Unicode escapes.
+  for (var i = 0; i < text.length;) {
+    var charsOnLine = nextRangeLength(text, i, 74);
+    buffer.write('\n    \'');
+    for (var c = 0; c < charsOnLine; c++) {
+      writeChar(buffer, text.codeUnitAt(i + c));
+    }
+    i += charsOnLine;
+    buffer.write('\'');
+  }
+  buffer.writeln(';');
+}
+
+// Code units that are encoded specially in the generated literals.
+const specialChars = {
+  0x08: r'\b',
+  0x09: r'\t',
+  0x0a: r'\n',
+  0x0b: r'\v',
+  0x0c: r'\f',
+  0x0d: r'\r',
+  0x24: r'\$',
+  0x27: r"\'",
+  0x7f: r'\x7F',
+  0x5c: r'\\',
+};
+
+/// Write a single code unit into a string literal.
+///
+/// Writes all code units above 0xFF as `\uHHHH`.
+/// Writes all printable ASCII codes as themselves,
+/// possibly escaped.
+/// Writes characters that have escapes, like `\n`, as those.
+/// The remaining code units in the 0x00..0xFF range are written as `\xHH`
+void writeChar(StringSink buffer, int charCode) {
+  if (specialChars[charCode] case var s?) {
+    buffer.write(s);
+  } else if (charCode >= 0x20 && charCode < 0x7f) {
```

Comment:
Instead of explicitly coding these character ranges, consider using the [category](https://en.wikipedia.org/wiki/Unicode_character_property#General_Category) of the unicode character to decide whether to escape it or not.

The script that I wrote only escaped characters in the "C" and "Z" categories, which results in this string for ISO-8859-6:

```
\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f\x10\x11\x12\x13
\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f\x20!"#$%&'()*+,-./0123456789:;
<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~
\x7f\x80\x81\x82\x83\x84\x85\x86\x87\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f\x90\x91
\x92\x93\x94\x95\x96\x97\x98\x99\x9a\x9b\x9c\x9d\x9e\x9f\xa0\ufffd\ufffd
\ufffd¤\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd،\xad\ufffd\ufffd\ufffd\ufffd
\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd؛\ufffd\ufffd\ufffd؟\ufffd
ءآأؤإئابةتثجحخدذرزسشصضطظعغ\ufffd\ufffd\ufffd\ufffd\ufffdـفقكلمنهوىيًٌٍَُِّْ\ufffd\ufffd\ufffd
\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd
```

It would also reduce your "special characters" map to just:
```dart
const specialChars = {
  0x27: r"\'",
  0x5c: r'\\',
}
```

Speaking of which, it is really hard to identify escaped characters in these strings. I'd suggest changing the quote character if the string contains a special character.

Reply:
I chose, perhaps not entirely necessarily, to keep the source ASCII, and encode any non-ASCII character.

I know Dart code is always UTF-8, but if I am reading this file at all, I prefer being able to tell the difference between "Α" and "A" .
(The former is upper-case Greek Alpha, U+0391, ISO-8859-7 char 0xC1, and the latter is the "plain" latin A, U+0041.)

The "ءآأؤإئابةتثجحخدذرزسشصضطظعغ" above also contains some right-to-left characters, which makes parsing the file even harder.

So, the options are:
* Optimize for size by using the actual code point for every character, not an escape.
* Optimize for size, to some degree, by using all *visible* characters, and escaping non-visible characters.
* Optimize for "readability", to some degree, by using all visible ASCII characters, and escaping the rest.
* Optimize for parsability by escaping everything with the same size escapes, `\u0061` for `A`.

Any option "works", because only the final `String` constant content matters.
I don't think size is relevant to optimize for. It's not a big download no matter what we do, and plain ASCII compresses well.

I ended up choosing the second to last option. Mainly because I was looking at the file myself while generating it, and that's the step I reached before I was satisfied.
But escaping everything would have been sufficient (and simpler, I wouldn't need to start counting entries per line for that.)

Having `\n`->`r'\n'` as a special character is really more of a gimmick. It's there, but ... nobody cares, because it's always in the same position in every table anyway. (And who even knows what `\f` and `\v` does. I looked it up.)

Reply:
I don't feel strongly about this. One advantage of using the actual characters instead of escapes is that you can visually see what types of characters your are encoding. But do this how you like. 

If you are interested in how other languages do this, Python uses your approach but has one character per line with a comment like:

```
    ...
    '\x1e'     #  0x1E -> RECORD SEPARATOR
    '\x1f'     #  0x1F -> UNIT SEPARATOR
    ' '        #  0x20 -> SPACE
    '!'        #  0x21 -> EXCLAMATION MARK
    '"'        #  0x22 -> QUOTATION MARK
    ...
```

Go does something similar.

One difference is that they do one code page per file so it is less obnoxious to use a lot of vertical space for each encoding.

Reply:
I think I have a slight preference for the the current approach of escaping non-ascii